### PR TITLE
refactor: improve dockerfiles

### PR DESCRIPTION
**Pull request by thde**

### CODE REVIEW

**Comment by jimmylevell** [comment]
LGTM. 

Only remark I have is the change of the docker base image: 
We especially specified it, so that 3rd party software (in this case dependabot) manage and update it. 

@diegosteiner : your thoughts?

File: .github/workflows/docker.yml [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    branches:
+      - main
+      - master
+  pull_request:
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    env:
+      DOCKER_CLI_EXPERIMENTAL: "enabled"
```
Comment (jimmylevell) [inline .github/workflows/docker.yml:14]: @thde : what is the reason for this env setting? Is it not deprecated? https://docs.docker.com/engine/deprecated/#:~:text=DOCKER_CLI_EXPERIMENTAL

**Review by jimmylevell — APPROVED** [review]
thank you for your great work :)

File: .github/workflows/docker.yml (hunk 1 above)
Comment (thde) [reply .github/workflows/docker.yml:14]: It is. There is no reason to keep it. Just copy-pasta... feel free to remove it.

**Comment by thde** [comment]
> LGTM.
> 
> Only remark I have is the change of the docker base image: We especially specified it, so that 3rd party software (in this case dependabot) manage and update it.
> 
> @diegosteiner : your thoughts?

Makes sense. Feel free to change it back to the initial versions. I can live with both :).

**Pull request merged**